packages/flutter/lib/src/services/image_resource.dart

```diff
@@ -16,13 +16,14 @@ typedef void ImageListener(ui.Image image);
 /// image object might change over time, either because the image is animating
 /// or because the underlying image resource was mutated.
 class ImageResource {
```

Comment:
I think the idiomatic thing to do, though, is to change the future from one that resolves a ui.Image to one that resolves to an immutable struct with a ui.Image and a scale.